test/features/test_organized_edge_detection.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#include <pcl/features/organized_edge_detection.h>
+#include <pcl/test/gtest.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+namespace {
+class OrganizedPlaneDetectionTestFixture : public ::testing::Test {
+protected:
+  const int INNER_SQUARE_EDGE_LENGTH = 50;
+  const int OUTER_SQUARE_EDGE_LENGTH = INNER_SQUARE_EDGE_LENGTH * 2;
+  const float SYNTHETIC_CLOUD_BASE_DEPTH = 2.0;
+  const float SYNTHETIC_CLOUD_DEPTH_DISCONTINUITY = .02f;
+  const float SYNTHETIC_CLOUD_RESOLUTION = 0.01f;
+
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud_;
+  pcl::PointIndicesPtr indices_;
+
+  void
+  SetUp() override
+  {
+    cloud_ = generateSyntheticEdgeDetectionCloud();
+  }
+
+private:
+  pcl::PointCloud<pcl::PointXYZ>::Ptr
+  generateSyntheticEdgeDetectionCloud()
+  {
+    auto organized_test_cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>>(
+        OUTER_SQUARE_EDGE_LENGTH, OUTER_SQUARE_EDGE_LENGTH);
+
+    // Draw a smaller square in front of a larger square both centered on the
+    // view axis to generate synthetic occluding and occluded edges based on depth
+    // discontinuity between neighboring pixels.  The base depth and resolution are
+    // arbitrary and useful for visualizing the cloud.  The discontinuity of the
+    // generated cloud must be greater than the threshold set when running the
+    // organized edge detection algorithm.
+    for (auto row = 0; row < OUTER_SQUARE_EDGE_LENGTH; ++row) {
+      for (auto col = 0; col < OUTER_SQUARE_EDGE_LENGTH; ++col) {
+        float x = col - (OUTER_SQUARE_EDGE_LENGTH / 2);
+        float y = row - (OUTER_SQUARE_EDGE_LENGTH / 2);
+
+        const auto outer_square_ctr = OUTER_SQUARE_EDGE_LENGTH / 2;
+        const auto inner_square_ctr = INNER_SQUARE_EDGE_LENGTH / 2;
```

Comment:
If you move this before  line 48, you can use `outer_square_ctr` in lines 48, 49.